### ISSUE TITLE
Shell function in Basic fails with 'file not found'

### ISSUE THREAD

**Issue by rebelxt** [flathub/org.libreoffice.LibreOffice#39]
LibreOffice flatpak version 6.0.4.2 in Linux Mint Mate 18.3:
I have a Basic function with the statement: Shell ("xed"). This statement fails with a 'file not found' error only in the flatpak version of LibreOffice. In case anybody wonders, Shell ("/usr/bin/xed") also fails.

**Comment by stbergmann** [comment]
That is because there is no xed executable bundled in the flatpak.  And it is more a security feature than a bug that the flatpak-sandboxed LO does not have access to all of the host environment, but just to select parts and in ways that are deemed secure.
One could imagine a portal through which a flatpak-sandboxed app could execute arbitrary shell-style commands in the host environment, if the user gives consent.  But nothing like that is planned for the flatpak infrastructure AFAIK.
In short, the Basic Shell command is, by design, only of limited use in the flatpak-sandboxed LO variant.

**Comment by nedrichards** [comment]
@stbergmann you could theoretically use HostCommand to run arbitary commands (GNOME Builder does) but that would need extra code and adaptation in LO. I agree that it should be outside the scope of the sandboxed build.
